senders/telegram/send.go

```diff
@@ -92,35 +118,108 @@ func (sender *Sender) buildMessage(events moira.NotificationEvents, trigger moir
 	return buffer.String()
 }
 
-func (sender *Sender) getChatUID(username string) (string, error) {
-	var uid string
-	if strings.HasPrefix(username, "%") {
-		uid = "-100" + username[1:]
-	} else {
-		var err error
-		uid, err = sender.DataBase.GetIDByUsername(messenger, username)
-		if err != nil {
-			return "", fmt.Errorf("failed to get username uuid: %s", err.Error())
-		}
+func (sender *Sender) getChat(contactValue string) (*Chat, error) {
+	var chat *Chat
+	var err error
+
+	switch {
+	// for private channel contactValue is transformed to be able to fetch it from telegram
+	case strings.HasPrefix(contactValue, "%"):
+		contactValue = "-100" + contactValue[1:]
+		chat, err = sender.getChatFromTelegram(contactValue)
+	// for public channel contactValue is transformed to be able to fetch it from telegram
+	case strings.HasPrefix(contactValue, "#"):
+		contactValue = "@" + contactValue[1:]
+		chat, err = sender.getChatFromTelegram(contactValue)
+	// for the rest of the cases (private chats, groups, supergroups), Chat data is stored in DB.
+	default:
+		chat, err = sender.getChatFromDb(contactValue)
 	}
-	return uid, nil
+
+	return chat, err
 }
 
-func (sender *Sender) getChat(username string) (*telebot.Chat, error) {
-	uid, err := sender.getChatUID(username)
+func (sender *Sender) getChatFromDb(contactValue string) (*Chat, error) {
+	var err error
+
+	chatRaw, err := sender.DataBase.GetIDByUsername(messenger, contactValue)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get username uuid: %s", err.Error())
```

Comment:
It is better to wrap errors using %w

Reply:
Done https://github.com/moira-alert/moira/pull/1022/commits/ebf30d12e8e97f8615b6bf842f27245e0aa05520